tests/cpp/ir/frontend_type_inference_test.cpp

```diff
@@ -47,5 +47,17 @@ TEST(FrontendTypeInference, BinaryOp) {
   EXPECT_EQ(truediv_f64->ret_type, PrimitiveType::f64);
 }
 
+TEST(FrontendTypeInference, UnaryOp) {
```

Comment:
FYI there's an `EXPECT_THROW` checker :-) https://google.github.io/googletest/reference/assertions.html#EXPECT_THROW

Reply:
Wow great! I'm just thinking about whether it is better to put the error checks in Python tests or C++ tests. Putting them in Python tests can directly give some illegal examples, while putting them in C++ tests can make `type_check` tests more uniform. WDYT?

Reply:
Haha this is a hard one, I'm more used to putting fine grained tests in C++ and keep python tests sparse but representable. (sorry I know this is vague...

Reply:
> putting fine grained tests in C++ and keep python tests sparse but representable

Sounds good. Will follow this in future PRs.